pkg/settlement/swap/swap_test.go

```diff
@@ -0,0 +1,505 @@
+	if err == nil {
+		t.Fatal("accepted invalid cheque")
+	}
+	if !errors.Is(err, swap.ErrWrongChequebook) {
+		t.Fatal("returned wrong error")
+	}
+
+	if observer.called {
+		t.Fatal("observer was be called for rejected payment")
+	}
+}
+
+func TestPay(t *testing.T) {
+	logger := logging.New(ioutil.Discard, 0)
+	store := mockstore.NewStateStore()
+
+	amount := uint64(50)
+	beneficiary := common.HexToAddress("0xcd")
+	var cheque chequebook.SignedCheque
+
+	peer := swarm.MustParseHexAddress("abcd")
+	var chequebookCalled bool
+	chequebookService := mockchequebook.NewChequebook(
+		mockchequebook.WithChequebookIssueFunc(func(b common.Address, a *big.Int, sendChequeFunc chequebook.SendChequeFunc) error {
+			if b != beneficiary {
+				t.Fatalf("issuing cheque for wrong beneficiary. wanted %v, got %v", beneficiary, b)
+			}
+			if a.Uint64() != amount {
+				t.Fatalf("issuing cheque with wrong amount. wanted %d, got %d", amount, a)
+			}
+			chequebookCalled = true
+			return sendChequeFunc(&cheque)
+		}),
+	)
+
+	networkID := uint64(1)
+	addressbook := &addressbookMock{
+		beneficiary: func(p swarm.Address) (common.Address, bool, error) {
+			if !peer.Equal(p) {
+				t.Fatal("querying beneficiary for wrong peer")
+			}
+			return beneficiary, true, nil
+		},
+	}
+
+	var emitCalled bool
+	swap := swap.New(
+		&swapProtocolMock{
+			emitCheque: func(ctx context.Context, p swarm.Address, c *chequebook.SignedCheque) error {
+				if !peer.Equal(p) {
+					t.Fatal("sending to wrong peer")
+				}
+				if !cheque.Equal(c) {
+					t.Fatal("sending wrong cheque")
+				}
+				emitCalled = true
+				return nil
+			},
+		},
+		logger,
+		store,
+		chequebookService,
+		mockchequestore.NewChequeStore(),
+		addressbook,
+		networkID,
+	)
+
+	err := swap.Pay(context.Background(), peer, amount)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !chequebookCalled {
+		t.Fatal("chequebook was not called")
+	}
+
+	if !emitCalled {
+		t.Fatal("swap protocol was not called")
+	}
+}
+
+func TestPayIssueError(t *testing.T) {
+	logger := logging.New(ioutil.Discard, 0)
+	store := mockstore.NewStateStore()
+
+	amount := uint64(50)
+	beneficiary := common.HexToAddress("0xcd")
+
+	peer := swarm.MustParseHexAddress("abcd")
+	reject := errors.New("reject")
+	chequebookService := mockchequebook.NewChequebook(
+		mockchequebook.WithChequebookIssueFunc(func(b common.Address, a *big.Int, sendChequeFunc chequebook.SendChequeFunc) error {
+			return reject
+		}),
+	)
+
+	networkID := uint64(1)
+	addressbook := &addressbookMock{
+		beneficiary: func(p swarm.Address) (common.Address, bool, error) {
+			if !peer.Equal(p) {
+				t.Fatal("querying beneficiary for wrong peer")
+			}
+			return beneficiary, true, nil
+		},
+	}
+
+	swap := swap.New(
+		&swapProtocolMock{},
+		logger,
+		store,
+		chequebookService,
+		mockchequestore.NewChequeStore(),
+		addressbook,
+		networkID,
+	)
+
+	err := swap.Pay(context.Background(), peer, amount)
+	if err == nil {
+		t.Fatal("expected error")
+	}
```

Comment:
This check is covered with the next one.

Reply:
fixed